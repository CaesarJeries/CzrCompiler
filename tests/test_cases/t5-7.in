void main()
{
    // break test
    while(true)
    {
        print("starting infinite loop\n");
        break;
    }
    print("out of loop through break statement\n");

    int x = 1;
    byte y = 255b;
    int count = 0;
    while(y/x >=1)
    {
        x = x + 2;
        count = count+1;
        if(count>3)
        {
            print("break\n");
            break;
        }
    }
    printi(x);
    print("\n");
    int z = y/x;
    printi(z);
    print("\n");
    x = 10;
    y = 10b;
    while(true)
    {
        print("before inner loop\n");
        while(x==y)
        {
            print("break inner loop\n");
            break;
        }

        print ("break outer loop\n");
        break;
    }
    print("finished");

}