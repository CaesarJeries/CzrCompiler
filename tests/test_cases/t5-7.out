Loaded: ./exceptions.s
starting infinite loop
out of loop through break statement
break
9
28
before inner loop
break inner loop
break outer loop
finished